### PR TITLE
Allow mock conductor to return user-specified errors

### PR DESCRIPTION
**Pull request by timotree3**

### CODE REVIEW

**Comment by timotree3** [comment]
This doesn't work for zome calls one sec

File: src/index.js [hunk 1]
```diff
@@ -65,15 +65,15 @@ class MockHolochainConductor {
     })
   }
 
-  any (response) {
-    this.anyResponse = response
+  any (response, returnError = false) {
```
Comment (robbiecarlton) [inline src/index.js:68]: I think I'd like the second arg in these calss to be an opts field, partly for future proofing, partly for readability in the calling code.

`mockHolochainConductor.next("error message", { returnError: true })`

is more readable than

`mockHolochainConductor.next("error message", true)`

(obv this applies to `next` and `once` as well)

**Review by robbiecarlton — CHANGES_REQUESTED** [review]
One requested API change, also can you update changelog and readme?

File: src/index.js (hunk 1 above)
Comment (timotree3) [reply src/index.js:68]: agreed

**Review by robbiecarlton — APPROVED** [review]
Looks good :shipit:

**Pull request merged**